spec/lib/elasticity/aws_request_spec.rb

```diff
@@ -145,10 +142,10 @@
 
     it 'should POST a properly assembled request' do
       ruby_params = {}
-      aws_params = {}
       Elasticity::AwsRequest.should_receive(:convert_ruby_to_aws).with(ruby_params).and_return(ruby_params)
-      request.should_receive(:sign_params).with(aws_params).and_return('SIGNED_PARAMS')
-      RestClient.should_receive(:post).with('PROTOCOL://HOSTNAME', 'SIGNED_PARAMS', :content_type => 'application/x-www-form-urlencoded; charset=utf-8')
+      request.should_receive(:headers).and_return('HEADERS')
+      request.should_receive(:payload).and_return('PAYLOAD')
+      RestClient.should_receive(:post).with('PROTOCOL://HOSTNAME', 'PAYLOAD', 'HEADERS')
```

Comment:
Can you explain the changes in this file? I'm really a loss, here.

Reply:
This file contains unit tests. I updated a few of the tests so that they pass with these changes. Then, I realized that most of the tests are broken in master. So I decided against trying to fix _all_ of the tests, but I left the ones that I was able to fix.